docs/table/mixin_columns.rst

```diff
@@ -0,0 +1,262 @@
+mixins within a masked table.  In this case a ``mask`` attribute is assigned to
+the mixin column object.  This ``mask`` is a special object that is a boolean
+array of ``False`` corresponding to the mixin data shape.  The ``mask`` looks
+like a normal numpy array but an exception will be raised if ``True`` is assigned
+to any element.  The consequences of the limitation are most obvious in the
+high-level table operations.
+
+**High-level table operations**
+
+The table below gives a summary of support for high-level operations on tables
+that contain mixin columns:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 28 72
+
+   * - Operation
+     - Support
+   * - :ref:`grouped-operations`
+     - Not implemented yet, but no fundamental limitation
+   * - :ref:`stack-vertically`
+     - Not implemented yet, pending definition of generic concatenation protocol
+   * - :ref:`stack-horizontally`
+     - Works if output mixin columns do not require masking
+   * - :ref:`table-join`
+     - Works if output mixin columns do not require masking; no mixin key
+       columns allowed
+   * - :ref:`unique-rows`
+     - Not implemented yet, uses grouped operations
+
+**Mixin column attributes**
+
+For mixin columns the column attributes ``name``, ``unit``, ``dtype``,
+``format``, ``description`` and ``meta`` are currently stored in a simple
+dictionary called `_astropy_column_attrs`.  These attributes can be manipulated
+with the functions ``col_getattr`` and ``col_setattr`` which are available in
+the ``astropy.table.column`` module.  These methods are not formally part of
+the astropy public API and may change in the future.
+
+
+Mixin protocol
+^^^^^^^^^^^^^^
+
+A key idea behind mixin columns is that any class which satisfies a specified
+protocol can be used.  That means many user-defined class objects which handle
+array-like data can be used natively within a |Table|.  The protocol is
+relatively simple and requires that a class behave like a minimal numpy array
+with the following properties:
+
+- Contains array-like data
+- Supports getting data as a single item, slicing, or index array access
+- Has a ``shape`` attribute
+- Has a ``__len__`` method for length
+- Has a ``_astropy_column_attrs`` attribute, which tells the |Table| class to
+  use the object natively instead of converting to a |Column|
+
+The `Example: ArrayWrapper`_ section shows a working minimal example of a class
+which can be used as a mixin column.  A `pandas.Series
+<http://pandas.pydata.org/pandas-docs/stable/generated/pandas.Series.html>`_
+object can function as a mixin column as well. 
+
+Other interesting possibilities for mixin columns include:
+
+- Columns which are dynamically computed as a function of other columns (AKA
+  spreadsheet)
+- Columns which are themselves a |Table|, i.e. nested tables
+
+.. _arraywrapper_example:
+
+Example: ArrayWrapper
+^^^^^^^^^^^^^^^^^^^^^
+
+The code listing below shows a example of a data container class which acts as
```

Comment:
Also, the doctests should run to make sure this works, so I'm also fine with leaving this in the docs.